### PR TITLE
Afform - reset managed entities when deleting a dashlet

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Fixes [dev/core#3122](https://lab.civicrm.org/dev/core/-/issues/3122)

Before
----------------------------------------
Deleting an afform does not delete the related dashlet

After
----------------------------------------
Now it does

Technical Details
----------------------------------------
It wasn't as simple as adding a cache clear in the revert function, because if multiple afforms were being reverted at once, that would be very slow clearing the cache each time. So I refactored the `BasicBatchAction` to allow the cache to be cleared after the whole batch is processed.

### CODE REVIEW

File: CRM/Core/ManagedEntities.php [hunk 1]
```diff
@@ -667,11 +667,11 @@ protected function loadDeclarations(): void {
 
   protected function loadManagedEntityActions(): void {
     $managedEntities = Managed::get(FALSE)->addSelect('*')->execute();
+    $this->managedActions = [];
```
Comment (colemanw) [inline CRM/Core/ManagedEntities.php:670]: Unit test revealed this variable was leaky. The purpose of this function is to rebuild this array, but it needs to be reset first due to the singleton pattern.

**Comment by colemanw** [comment]
@MegaphoneJon can you confirm this fixes your issue?

**Comment by colemanw** [comment]
@MegaphoneJon not according to the unit tests.This is still passing:

https://github.com/civicrm/civicrm-core/blob/095e8ae41f6c71a8c7372828046fc012545b85a6/ext/search_kit/tests/phpunit/api/v4/SearchDisplay/SearchAfformTest.php#L242-L247

**Comment by colemanw** [comment]
@MegaphoneJon actually I just took a closer look and you're right. The unit test wasn't catching this problem with deleting an afform using a "default display": https://github.com/civicrm/civicrm-core/pull/22997

So if we merge that PR as well as this one, it should all be working well according to the tests.

**Comment by colemanw** [comment]
@MegaphoneJon now that #22997 is merged are you good with merging this one too?

**Comment by MegaphoneJon** [comment]
I have no objection to this being merged - but it also seems as though #22997 was the bug that was generating the error for end users.

When this PR is applied I can see that the entry in `civicrm_dashboard` is deleted immediately when the Search Kit search is deleted.  However, when the patch isn't applied, the dashboard isn't broken - it simply doesn't show the dashlet, and when cache is cleared, the `civicrm_dashboard` record is removed.

**Pull request merged**